app/src/app.rs

```diff
@@ -232,18 +230,19 @@ impl Component for App {
     }
 
     fn view(&self) -> Html {
+        let link = self.link.clone();
         let candidates = Rc::clone(&self.candidates);
+        let candidates_selection = self.candidates_selection.clone();
         let entries = Rc::clone(&self.entries);
-        let selected_timezone = self.selected_timezone.clone();
-        let tz_range = Duration::hours(TZ_RANGE);
         let collapsed = self.collapsed.clone();
         let show_contractor = self.show_contractor.clone();
         let show_employee = self.show_employee.clone();
         let show_pending = self.show_pending.clone();
         let show_select = self.show_select.clone();
         let show_unselect = self.show_unselect.clone();
-        let link = self.link.clone();
-        let candidates_selection = self.candidates_selection.clone();
+        let offset_bounds = UtcOffsetBounds::new(3).clone();
+        let selected_timezone = self.selected_timezone.clone();
+        let offset_ranges = UtcOffsetRanges::new(selected_timezone, offset_bounds.clone()).clone();
```

Comment:
There doesn't seem to be a reason to clone here. clippy didn't tell you?

The reason we clone for the others is because they are taken from `self` which is a reference.

```suggestion
        let offset_bounds = UtcOffsetBounds::new(3);
        let selected_timezone = self.selected_timezone.clone();
        let offset_ranges = UtcOffsetRanges::new(selected_timezone, offset_bounds.clone());
```